handler/src/main/java/io/netty/handler/ssl/SslHandler.java

```diff
@@ -412,22 +412,30 @@ public String applicationProtocol() {
      *
      * @return the {@link Future} for the iniital TLS handshake if {@link #renegotiate()} was not invoked.
      *         The {@link Future} for the most recent {@linkplain #renegotiate() TLS renegotiation} otherwise.
+     * @deprecated intercept {@link SslHandshakeCompletionEvent}.
      */
+    @Deprecated
     public Future<Channel> handshakeFuture() {
```

Comment:
@Scottmitch let us just only deprecate close for now and then we can re-evaluate if needed. Ok ?